src/v2/manifest/mod.rs

```diff
@@ -141,6 +105,63 @@ impl Client {
         }
     }
 
+    async fn build_url(&self, name: &str, reference: &str) -> Result<Url> {
+        let ep = format!(
+            "{}/v2/{}/manifests/{}",
+            self.base_url.clone(),
+            name,
+            reference
+        );
+        match reqwest::Url::parse(&ep) {
+            Ok(url) => Ok(url),
+            Err(e) => {
+                return Err(Error::from(format!(
+                    "failed to parse url from string '{}': {}",
+                    ep, e
+                )));
+            }
+        }
+    }
+
+    /// Fetch content digest for a particular tag
+    pub async fn get_manifestref(&self, name: &str, reference: &str) -> Result<Option<String>> {
+        let url = self.build_url(name, reference).await?;
+
+        let accept_headers = build_accept_headers().await;
+
+        let res = self
+            .build_reqwest(
+                reqwest::Client::new()
+                    .head(url.clone())
```

Comment:
If this `url` is not used further on (it doesn't seem so), I think you don't need to `clone()` it here.